build.gradle.kts

```diff
@@ -125,16 +128,23 @@ tasks.register<Jar>("sourcesJar") {
     from(sourceSets.named("main").get().allJava)
     classifier = "sources"
 }
-
 tasks.register<Jar>("javadocJar") {
     from(tasks.named<Javadoc>("javadoc"))
     classifier = "javadoc"
 }
-
+tasks.named<Javadoc>("javadoc") {
+    if (JavaVersion.current().isJava9Compatible) {
+        (options as StandardJavadocDocletOptions).addBooleanOption("html5", true)
+    }
+}
 tasks.register<Jar>("testJar") {
     from(sourceSets.named("junitTest").get().output)
     classifier = "tests"
 }
+tasks.register<Zip>("distZip") {
+    from("$buildDir/dist")
+}
+tasks.getByName("distZip").dependsOn("dist")
```

Comment:
avoid `getByName` it is eagerly creating task, see guidelines https://docs.gradle.org/current/userguide/task_configuration_avoidance.html